experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -158,62 +146,88 @@ function rebaseMarkList<TNodeChange>(
 		// Note that we first add lineage for `baseMark` to `lineageRecipients`, then handle adding lineage to `rebasedMark`,
 		// then add `baseMark` to `lineageEntries` so that `rebasedMark` does not get an entry for `baseMark`.
 		// `rebasedMark` should already have a detach event for `baseMark`.
-		if (markEmptiesCells(baseMark)) {
-			assert(isDetachMark(baseMark), 0x709 /* Only detach marks should empty cells */);
-			const detachId = getDetachCellId(baseMark, baseIntention);
+		if (markEmptiesCells(baseMark) || isTransientEffect(baseMark)) {
+			const detachId = getOutputCellId(baseMark, baseRevision, metadata);
+			assert(detachId !== undefined, "Mark which empties cells should have a detach ID");
 			assert(detachId.revision !== undefined, 0x74a /* Detach ID should have a revision */);
-			addLineageToRecipients(lineageRecipients, detachId.revision, detachId.localId, length);
-		}
+			addLineageToRecipients(
+				lineageRecipients,
+				detachId.revision,
+				detachId.localId,
+				baseMark.count,
+				metadata,
+			);
 
-		if (areInputCellsEmpty(rebasedMark)) {
-			if (markEmptiesCells(baseMark)) {
-				assert(isDetachMark(baseMark), 0x74b /* Only detaches empty cells */);
-				if (baseMark.type === "MoveOut" || baseMark.detachIdOverride === undefined) {
-					setMarkAdjacentCells(rebasedMark, detachBlock);
-				}
+			if (isInverseAttach(baseMark)) {
+				assert(detachId.revision !== undefined, 0x74c /* Detach ID should have revision */);
+				lineageEntries.push({
+					revision: detachId.revision,
+					id: detachId.localId,
+					count: baseMark.count,
+				});
 			} else {
-				handleLineage(
+				addIdRange(getOrAddEmptyToMap(detachBlocks, detachId.revision), {
+					id: detachId.localId,
+					count: baseMark.count,
+				});
+			}
+
+			assert(
+				areInputCellsEmpty(rebasedMark) && rebasedMark.cellId.revision !== undefined,
+				"Mark should have empty input cells after rebasing over a cell-emptying mark",
+			);
```

Comment:
Nit: if you move this asser above the `if (isInverseAttach(baseMark)) {` block, you can merge the `else` block with the `if (!isInverseAttach(baseMark)) {` block below